patches/server/0962-Piglin-Dancing-API.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Amin <amin.haddou@frg.wwschool.de>
+Date: Wed, 22 Feb 2023 10:27:04 +0100
+Subject: [PATCH] Piglin Dancing API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPiglin.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPiglin.java
+index 48d0a4e42e1b90d1323784d1284acabfe9497dd6..b3b0d58082d39575788d93ba7c787576a455b7c1 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPiglin.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPiglin.java
+@@ -100,5 +100,27 @@ public class CraftPiglin extends CraftPiglinAbstract implements Piglin, com.dest
+     public boolean isChargingCrossbow() {
+         return this.getHandle().isChargingCrossbow();
+     }
++
++    @Override
++    public void setDancing(boolean dancing) {
++        if (dancing) {
++            this.getHandle().getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.DANCING, true);
++            this.getHandle().getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.CELEBRATE_LOCATION, this.getHandle().getOnPos());
++        } else {
++            this.getHandle().getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.DANCING);
++            this.getHandle().getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.CELEBRATE_LOCATION);
++        }
++    }
++
++    @Override
++    public void setDancing(int duration) {
++        this.getHandle().getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.DANCING, true);
```

Comment:
actually, one more thing, I know nothing about the brain api, but why do you set one with an expiry and one without?

Reply:
There isn't a particular reason for that, I changed it now so that both memories expire